topics/reference_guide/custom_language_support/websymbols_implementation.md

```diff
@@ -0,0 +1,384 @@
+<!-- Copyright 2000-2023 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file. -->
+# Implementing Web Symbols
+
+<link-summary>
+Implementation details for the Web Symbols API.
+</link-summary>
+
+The core element of the framework is a
+[`WebSymbol`](%gh-ic%/platform/webSymbols/src/com/intellij/webSymbols/WebSymbol.kt),
+which represents an entity in the Web Symbols model.
+It is described through `namespace`, `kind` and `name` properties. Its lifecycle is a single read action.
+If you need it to survive between read actions, use `WebSymbol.createPointer()` to create a symbol pointer.
+If the symbol is still valid, dereferencing the pointer will return a new instance of the symbol.
+
+The property `namespace` describes which language or concept the symbol belongs to,
+and `kind` describes which group of symbols within that particular language or concept it belongs to.
+E.g. a CSS property would have: `namespace == "CSS"` and `kind == "properties"`, a Java class would have:
+`namespace == "Java"` and `kind == "classes"`.
+The namespace can also represent a concept, which is not tied to a particular language.
+For instance a plugin extension could be represented with `namespace == "ij-plugin"` and `kind == "extensions"`.
+
+A Web Symbol can originate from source code analysis or it can be a symbol statically defined
+through Web Types (JSON) or some other, custom format (see [](websymbols_static_symbols.md)).
+In both cases such a symbol can have some `source` defined.
+Each symbol is treated by the framework the same, regardless of their origin.
+
+## General Properties
+
+WebSymbol has a number of properties, which are used across IDE features:
+
+{style="narrow"}
```

Comment:
Unfortunately, the narrow style doesn't fit many of the property/method names.

Please remove it everywhere, and let's use the default one. I think we will end up moving these descriptions to the code anyway in the future.

Reply:
👍🏻